### PR TITLE
Generate schema spec into the site documentation

### PR DESCRIPTION
**Pull request by martina-if**
- [x] Code compiles correctly (i.e `make build`)
- [x] All unit tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested

### CODE REVIEW

File: pkg/schema/generate.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const outpufFile = "site/content/usage/20-schema.md"
+
+func main() {
+
+	var document strings.Builder
+	document.WriteString("---\n")
+	document.WriteString("title: Config file schema\n")
+	document.WriteString("weight: 200\n")
+	document.WriteString("---\n\n")
```
Comment (cPu1) [inline pkg/schema/generate.go]: Minor nit: we could also use a template string and a single `WriteString` call for the content till here.

```golang
	b.WriteString(`---
title: Config file schema
weight: 200
---

`)
```

File: pkg/schema/generate.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	document.WriteString("```yaml\n")
+
+	schema := jsonschema.Reflect(&api.Config{})
+	yamlSchema, _ := yaml.Marshal(schema.Definitions)
+	document.WriteString(string(yamlSchema))
```
Comment (cPu1) [inline pkg/schema/generate.go]: Minor nit: the conversion to string can be avoided by using `document.Write`.

File: pkg/schema/generate.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"github.com/alecthomas/jsonschema"
+	"io/ioutil"
+	"k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/yaml"
+	"strings"
+)
+
+const outpufFile = "site/content/usage/20-schema.md"
```
Comment (cPu1) [inline pkg/schema/generate.go]: Minor typo: `outputFile`

File: pkg/schema/generate.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+	var document strings.Builder
+	document.WriteString("---\n")
+	document.WriteString("title: Config file schema\n")
+	document.WriteString("weight: 200\n")
+	document.WriteString("---\n\n")
+	document.WriteString("```yaml\n")
+
+	schema := jsonschema.Reflect(&api.Config{})
+	yamlSchema, _ := yaml.Marshal(schema.Definitions)
```
Comment (cPu1) [inline pkg/schema/generate.go]: Should we ignore the error? Looks like `errcheck` warns if the error isn't checked.

File: pkg/schema/generate.go (hunk 1 above)
Comment (martina-if) [reply pkg/schema/generate.go]: Good idea, I forget about these, thanks

**Comment by errordeveloper** [comment]
A couple of errors in CI:
```
pkg/schema/generate.go:4:2:error: cannot find package "github.com/alecthomas/jsonschema" in any of: (vet)
pkg/schema/generate.go:4:2:warning: error return value not checked (could not import github.com/alecthomas/jsonschema (cannot find package "github.com/alecthomas/jsonschema" in any of:) (errcheck)
```

File: pkg/schema/generate.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	var document strings.Builder
+	document.WriteString(`---
+title: Config file schema
+weight: 200
+---
+
+`)
+	document.WriteString("```yaml\n")
+
+	schema := jsonschema.Reflect(&api.Config{})
```
Comment (errordeveloper) [inline pkg/schema/generate.go]: Did you try it with our struct?

File: pkg/schema/generate.go [hunk 6]
```diff
@@ -0,0 +1,38 @@
+package main
```
Comment (errordeveloper) [inline pkg/schema/generate.go]: I'd say we should move it under `cmd/`, as we don't have `main` package in `pkg/` (aside from things that get have build tags and only used with `go generate`, IIRC).

File: pkg/schema/generate.go (hunk 5 above)
Comment (martina-if) [reply pkg/schema/generate.go]: yeah, I used the k8s one by mistake. Thanks!

**Pull request merged**